das/checkpoint.go

```diff
@@ -0,0 +1,54 @@
+package das
+
+import (
+	"bytes"
+	"fmt"
+)
+
+type checkpoint struct {
+	SampledBefore uint64             `json:"sampled_before"`
+	MaxKnown      uint64             `json:"max_known"`
```

Comment:
These two need better names since they are logged out. 
I suggest
`MaxKnown` => `LatestSampledHeader`
`SampledBefore` => `HeadOfSampledChain` (something to indicate that it is the tip of the contiguously successfully sampled chain of headers

Reply:
Nice suggestion! Maybe do it a bit differently:
 - `MaxKnown` => `Head`: this header could be not sampled yet, so ` LatestSampledHeader`  could possible be misleading. On the other hand it is height of most recent Head so maybe just rename it to `Head`?
 - `SampledBefore` => `LatestSampledHeader` I think this is good name for it since all header before this one were sampled.

Reply:
MaxKnown => NetworkHead
SampledBefore => HeadOfSampledChain in user interface
SampledBefore => SampleFrom in checkpoint